File: series0_handout/series0_handout/midpoint/plot_error_compare.py
import numpy
import matplotlib
import matplotlib.pyplot as plt
errors = numpy.loadtxt('series0_1_d_errors.txt')
numberOfIntervals = numpy.loadtxt('series0_1_d_numbers.txt')

plt.loglog(numberOfIntervals, errors, '-*', label='$|I_n(f)-I(f)|$')
plt.title('Error plot for the midpoint rule')
plt.xlabel('$n$')
plt.ylabel('$|I(f)-I_n(f)|$')
plt.grid('on')

# Now compare against the rate n^(-2)
plt.loglog(numberOfIntervals, numberOfIntervals**(-2), '--',label='$\mathcal{O}(n^{-2})$')
plt.legend()
plt.show()

File: series0_handout/series0_handout/midpoint/plot_error_compare.m
errors = load('series0_1_d_errors.txt');
numberOfIntervals = load('series0_1_d_numbers.txt');

loglog(numberOfIntervals, errors, '*-');
title('Error plot for ex. 0.1d)')
xlabel('n')
ylabel('|I(f)-I_n(f)|')
hold on
loglog(numberOfIntervals, numberOfIntervals.^(-2), '--')

legend('|I_n(f)-I(f)|', 'O(n^{-2})')
grid on
grid minor
